code/harness/DiskContents.cpp

```diff
@@ -0,0 +1,298 @@
+#include "DiskContents.h"
+using std::endl;
+
+namespace fs_testing {
+
+fileAttributes::fileAttributes() {
+  dir_attr = NULL;
+  stat_attr = NULL;
+  md5sum = "";
+}
+
+fileAttributes::~fileAttributes() {
```

Comment:
You leak the `dir_attr` and `stat_attr` fields that you allocate below, I would recommend just embedding the structs in the class (so `struct dirent dir_attr` instead of `struct dirent *dir_attr`)